pages/price-feeds/use-real-time-data/fuel.mdx

```diff
@@ -0,0 +1,188 @@
+
+<Callout type="info" emoji="ℹ️">
+  Regarding the Pyth contract on Fuel, the caller must pay the fee in the base
+  asset for functions like `update_price_feeds`. The fee is currently set to the
+  minimum possible value (1 wei).
+</Callout>
+
+The code snippet above does the following things:
+
+1. Defines an `UpdatePrice` ABI with functions to interact with the Pyth contract.
+2. Implements the `UpdatePrice` ABI for the contract, providing the following functionality:
+
+   - `valid_time_period()`: Retrieves the valid time period from the Pyth contract.
+   - `get_price(price_feed_id)`: Gets the price for a given price feed ID.
+   - `get_price_unsafe(price_feed_id)`: Gets the price for a given price feed ID without staleness checks.
+   - `update_fee(update_data)`: Calculates the fee required to update the price feeds.
+   - `update_price_feeds(update_fee, update_data)`: Updates the price feeds with the provided data and fee.
+
+3. Uses the `PYTH_CONTRACT_ID` constant to interact with the Pyth contract on testnet.
+4. Uses the `FUEL_ETH_BASE_ASSET_ID` constant as the asset ID for paying update fees.
+
+To use this contract, you would typically:
+
+1. Call `update_fee()` to get the required fee for updating price feeds.
+2. Call `update_price_feeds()` with the fee and update data to refresh the price feeds.
+3. Use `get_price()` or `get_price_unsafe()` to read the updated prices.
+
+### Write Client Code
+
+The code snippet below provides an example of how to fetch price updates using NextJS, a full example can be found [here](https://github.com/pyth-network/pyth-examples/tree/main/price_feeds/fuel/fetch-and-update-btc-price).
+
+```ts copy
+import { TestContractAbi__factory } from "@/sway-api";
+import PYTH_CONTRACT_ABI from "../abi/pyth-contract-abi.json";
+import { arrayify, Contract, hexlify } from "fuels";
+
+const HERMES_V2_LATEST_PRICE_UPDATES_ENDPOINT =
+  "https://hermes.pyth.network/v2/updates/price/latest?ids[]=";
+const FUEL_ETH_BASE_ASSET_ID =
+  "0xf8f8b6283d7fa5b672b530cbb84fcccb4ff8dc40f8176ef4544ddb1f1952ad07";
+const PRICE_FEED_ID =
+  "0xff61491a931112ddf1bd8147cd1b641375f79f5825126d665480874634fd0ace"; // ETH/USD
+
+const contractId =
+  CURRENT_ENVIRONMENT === "local"
+    ? contractIds.testContract
+    : (process.env.NEXT_PUBLIC_TESTNET_CONTRACT_ID as string); // Testnet Contract ID
+const pythContractId = process.env
+  .NEXT_PUBLIC_PYTH_TESTNET_CONTRACT_ID as string; // Testnet Contract ID
+
+async function updateAndGetPrice() {
+  const fetchPriceUpdateData = async () => {
+    const response = await fetch(
+      HERMES_V2_LATEST_PRICE_UPDATES_ENDPOINT + PRICE_FEED_ID
+    );
+    if (!response.ok) {
+      throw new Error("Failed to fetch price");
+    }
+    const data = await response.json();
+    const binaryData = data.binary.data[0];
+    const buffer = Buffer.from(binaryData, "hex");
+    return buffer;
+  };
+
+  const updateData = await fetchPriceUpdateData();
```

Comment:
Why are you not using the hermes client here?

Reply:
we could but this was just more straightforward